Better docs (#58)
Prividema/prividema-fhe: BUILDING.md

# How to build Prividema-fhe

## Requirements

- CMake version 3.10

> - [Criterion](https://github.com/Snaipe/Criterion) if you want to if you want to run the unit tests. In this case make sure that pkg-config is also installed for portability between different OS.

Reviewer: ```suggestion
- [Criterion](https://github.com/Snaipe/Criterion) optional, only for running the unit tests. In this case make sure that pkg-config is also installed for portability between different OS.
```

        cd Criterion/
        mkdir build
        meson setup build
        sudo meson install -C build

 1. Verify installation:

        ls /usr/include/criterion
        ls /usr/lib/libcriterion*

 Notes:

- Linking math library (-lm) may be required on Linux

##### MAC USERS

 1. Install Criterion using Homebrew (recommended):

        brew install criterion

 2. Ensure Meson is installed (required by Criterion):

        brew install meson

 3. Verify installation:

     _Apple Silicon (arm64)_

        ls /opt/homebrew/include/criterion
        ls /opt/homebrew/lib/libcriterion*

    _Intel macOS_

        ls /usr/local/include/criterion
        ls /usr/local/lib/libcriterion*

 4. Optional: build from source

        git clone <https://github.com/Snaipe/Criterion.git>
        cd Criterion/
        mkdir build
        meson setup build
        meson install -C build

#### Running the tests

It might be helpful to know that

```bash
ctest
```

Inside the build directory will automatically run all tests.
Make sure that you have abided by the above warning, as ctest will not by default
tell you how many tests have been run in each file (and you might thing that everything

> passes when in reality no tests are being run at all).

Reviewer: ```suggestion
tell you how many tests have been run in each file (one might think that everything
passes when in reality no tests are being run at all).
```
